internal/repository/sync.go

```diff
@@ -64,8 +68,49 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
```

Comment:
Why must this be sorted?

Reply:
I see why, perhaps this can use the set implementation in #304 instead. :thinking:

Reply:
We cannot use the set implementation, since we would have to be using `*tuf.Key`'s as keys, but since they are pointers, when checking if the set contains a key, it might think that the key is not their, when it actually is, since the exact "key" is not their, but a key with the same values is their. We cannot use the not pointer version of `*tuf.Key` since it is a complex structure